verilog/analysis/lint_rule_registry.h

```diff
@@ -98,14 +96,16 @@ using LintRuleDescriptionsMap =
 //   return "my-lint-rule";  // safely initialized function-local string literal
 // }
 //
+// TODO(hzeller): once the class does not contain a state, extract the name
+// and description from the instance to avoid weird static initialization.
 #define VERILOG_REGISTER_LINT_RULE(class_name)                               \
   static verilog::analysis::LintRuleRegisterer<class_name::rule_type>        \
       __##class_name##__registerer(                                          \
-          class_name::Name(),                                                \
+          class_name::GetDescriptor().name, /* todo: only call once */       \
```

Comment:
Or is this redundant from the argument passed a few lines below?

Reply:
Ah, yeah, that was what I changed right in the beginning and forgot the TODO. Changed.